### ISSUE TITLE
Fix -Warray-bounds issues in drivers/platform/chrome/cros_ec_vbc.c

### ISSUE THREAD

**Issue by GustavoARSilva** [KSPP/linux#278]
Seen under GCC-13

```c
drivers/platform/chrome/cros_ec_vbc.c: In function ‘vboot_context_read’:
drivers/platform/chrome/cros_ec_vbc.c:36:15: warning: array subscript ‘struct ec_params_vbnvcontext[1]’ is partly outside array bounds of ‘unsigned char[36]’ [-Warray-bounds=]
   36 |         params->op = EC_VBNV_CONTEXT_OP_READ;
      |               ^~
In file included from drivers/platform/chrome/cros_ec_vbc.c:12:
In function ‘kmalloc’,
    inlined from ‘vboot_context_read’ at drivers/platform/chrome/cros_ec_vbc.c:30:8:
./include/linux/slab.h:580:24: note: at offset 20 into object of size 36 allocated by ‘kmalloc_trace’
  580 |                 return kmalloc_trace(
      |                        ^~~~~~~~~~~~~~
  581 |                                 kmalloc_caches[kmalloc_type(flags)][index],
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  582 |                                 flags, size);
      |                                 ~~~~~~~~~~~~
```

**Comment by GustavoARSilva** [comment]
Patch: https://lore.kernel.org/linux-hardening/ZCTrutoN+9TiJM8u@work/

**Comment by kees** [comment]
Fixed in commit 59a9ccf19ee03179faf047822bbec76cac7467a4.
